<launch>
1;4205;0c  <arg name="spi_id" default="1.0"/>
  <arg name="i2c_id" default="0"/>
  <arg name="frame_id" default="lepton_camera"/>
  <arg name="headless" default="true"/>

  <group ns="$(arg frame_id)">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" output="screen" />


    <node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load lepton_camera_driver/LeptonThermalSensorNodelet camera_nodelet_manager" output="screen">
      <param name="spi_dev" value="/dev/spidev$(arg spi_id)"/>
      <param name="i2c_dev" value="(arg i2c_id)"/>
      <param name="frame_id" value="$(arg frame_id)"/>
      <param name="camera_info_url" value="file://$(find lepton_thermal_sensor)/config/lepton_camera_parameter.yaml"/>
      <rosparam file="$(find lepton_thermal_sensor)/config/lepton_temperature_model.yaml" command="load" />
    </node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen" unless="$(arg headless)">
      <remap from="image" to="$(arg frame_id)/image_raw"/>
    </node>
  </group>

</launch>
